### PR TITLE
Upgrade to Bootstrap 5. Closes #238

### LINKED ISSUE #238: Upgrade to Bootstrap 4, including example of font-awesome-loader
Once this gets merged: https://github.com/shakacode/bootstrap-loader/pull/56

Then we can create a branch to update from Bootstrap 3 to Bootstrap 4.

We should also show an example of using font-awesome with the [font-awesome-loader](https://github.com/shakacode/font-awesome-loader).

### PR DESCRIPTION
**Pull request by vtamara**
* Replaces `bootstrap-loader` with Bootstrap 5 loaded directly with shakapacker
* Updates navigation bar to Bootstrap 5 and changes presentation to make it somehow similar to the Bootstrap 3 version
* Upgrades `react-bootstrap` to a version that handles Bootstrap 5 and adjusts css
* Footer improved minimally

After this PR, more work is needed in transition from one demo to another  (e.g From "React Router" to "Classic").  With this PR the transition redraws full page without style and after applies style, and it is noticeable but it was not noticeable with Bootstrap 3.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/498)

### CODE REVIEW

**Comment by vtamara** [comment]
https://user-images.githubusercontent.com/701221/160287805-be0071ed-eeef-4cb1-9f63-214642869261.mp4

File: Gemfile [hunk 1]
```diff
@@ -3,10 +3,10 @@
 source "https://rubygems.org"
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby "2.7.5"
+ruby ">=2.7.5"
```
Comment (justin808) [inline Gemfile:6]: That's a nice option

**Review by justin808 — CHANGES_REQUESTED** [review]
Nice work.

Is this PR ready to merge?

> Reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/498#-:-MzEO7CE5ENsjmX9Hxqx:bcamyws)* status: all files reviewed, 2 unresolved discussions (waiting on @vtamara)

___
*[client/app/bundles/comments/components/CommentBox/CommentForm/CommentForm.jsx, line 103 at r1](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/498#-MzENMETCxUfQoZ0YRh3:-MzENMETCxUfQoZ0YRh4:b6lgqlc) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/136f2041d657e95f8415b873abb82745b4641e12/client/app/bundles/comments/components/CommentBox/CommentForm/CommentForm.jsx#L103)):*
> ```JavaScript
> 
>     let ref;
>     switch (+this.state.formMode) {
> ```

what does this plus do?

**Comment by vtamara** [comment]
> Nice work.
> 
> Is this PR ready to merge?
> 
> > Reviewed 15 of 15 files at r1, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/498#-:-MzEO7CE5ENsjmX9Hxqx:bcamyws)_ status: all files reviewed, 2 unresolved discussions (waiting on @vtamara)
> 
> _[client/app/bundles/comments/components/CommentBox/CommentForm/CommentForm.jsx, line 103 at r1](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/498#-MzENMETCxUfQoZ0YRh3:-MzENMETCxUfQoZ0YRh4:b6lgqlc) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/136f2041d657e95f8415b873abb82745b4641e12/client/app/bundles/comments/components/CommentBox/CommentForm/CommentForm.jsx#L103)):_
> 
> > ```js
> >     let ref;
> >     switch (+this.state.formMode) {
> > ```
> 
> what does this plus do?

IMHO it is ready to merge.

The plus converts from string to integer in case that this.state.formMode is string (for some reason it was string in my case).

**Comment by justin808** [comment]
@Judahmeek merge?

**Comment by justin808** [comment]
@vaukalak Please review and merge if CI is passing.

**Comment by justin808** [comment]
@ahangarha @Judahmeek  I think it's worth considering merging this.

**Comment by ahangarha** [comment]
@vtamara Would you update your PR and resolve conflicts and CI failures?

**Comment by ahangarha** [comment]
@justin808 
Is this still wanted?

**Comment by justin808** [comment]
see #528

**Pull request closed without merging**